test/runtests.jl

```diff
@@ -111,5 +111,53 @@ using Test
             @test call1 != call3
             @test call2 != call4
         end
+
+        @testset "splatting" begin
+            f(a, b; kw1=1, kw2=2) = a + b + kw1 + kw2
+
+            one = 1
+            a = [1, 2]
+            kw = (kw1=1, kw2=2)
+
+            @testset "basic splatting" begin
+                @test f(a...) == @cached_call f(a...)
+                @test f(a...) == @cached_call f([1, 2]...)
+                @test f(a...) == @cached_call f([one, 2]...)
+                @test f(a...; kw...) == @cached_call f(a...; kw...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=1, kw2=2)...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=one, kw2=2)...)
+
+                @test @hash_call(f(a...)) isa UInt
+                @test @hash_call(f([1, 2]...)) isa UInt
+                @test @hash_call(f([one, 2]...)) isa UInt
+                @test @hash_call(f(a...; kw...)) isa UInt
+                @test @hash_call(f(a...; (kw1=1, kw2=2)...)) isa UInt
```

Comment:
I'm not sure how often it will come up and can probably be discounted,  but perhaps you can order the keyword arguments by name (or explicity document it) may want to do some argument ordering because: 

```
println(@hash_call(f(a...; (kw1=1, kw2=2)...))) # 10802886523120914904
println(@hash_call(f(a...; (kw2=1, kw1=2)...))) # 6706884303044505012
```

Reply:
Well noted. We don't make the promise in the API that different kwarg orders will hash to the same result, so I think it is not a bug but rather a feature request? I will make an issue.

Reply:
This actually is probably more pertinent example: Adding a keyword argument value, even if it's the same as the default, leads to a different hash: 

```
println(@hash_call(f(a...; (kw1=1, kw2=2)...))) # 10802886523120914904
println(@hash_call(f(a...; (kw2=1, kw1=2)...))) # 6706884303044505012
println(@hash_call(f(a...; (kw2=2)...))) # 5520464253110657799
```

Though this isn't really part of this MR.

Reply:
Yeah that's also right. I think in both cases the upshot is the same: Would be nice to have for efficiency but the worst case scenario is to just recompute the value 🤷.

Reply:
That said, the default values would be much harder to implement (I can't think of how to do it right now, as we would have to access the original definition of the function, not just the function call)